tests/test_documentation.py

```diff
@@ -0,0 +1,58 @@
+from inspect import isfunction, isclass, getmembers, ismodule, ismethod, signature, getmodule
+import importlib
+from itertools import compress
+import pytest
+
+from keras.utils.test_utils import keras_test
+
+modules = ['keras.layers', 'keras.models', 'keras', 'keras.backend.tensorflow_backend']
+accepted_name = ["from_config"]
+accepted_module = ["keras.legacy.layers", "keras.utils.generic_utils"]
+
+
+def handle_class(name, member):
+    if name in accepted_name or member.__module__ in accepted_module:
+        return
+    assert member.__doc__ is not None, "class doesn't have any documentation {} {}".format(name, member.__module__, getmodule(member).__file__)
+    for n, met in getmembers(member):
+        if ismethod(met):
+            handle_method(n, met)
+
+
+def handle_function(name, member):
+    if name in accepted_name or member.__module__ in accepted_module:
+        return
+    doc = member.__doc__
+    assert doc is not None, "{} function doesn't have any documentation {} {}".format(name, member.__module__, getmodule(member).__file__)
+    args = list(signature(member).parameters.keys())
```

Comment:
We should not only test for argument presence, we should test for:

- section presence, e.g. `# Returns`, IFF the function returns something, `# Raises` IFF exceptions are raised, `# Arguments` IFF any arguments
- argument order
- presence of a one-line description that starts on the same line as the docstring marker and ends with a dot.